### ISSUE TITLE
Deploy new national commute data

### ISSUE THREAD

**Issue by Robinlovelace** [npct/pct-shiny#703]
As discussed with @AnnaGoodman1 the data is now ready to go. The only data that will change is the vector route network. As of https://github.com/npct/pct-outputs-regional-R/commit/cb9f767773fda494de87d4590cef162ab32b221d we now have test data for the `isle-of-wight`, which we can use to develop on the test server. 

The plan for the new data is to:

- [x]  Create 2 files for each region, `rnet.Rds` and `rnet_full.Rds` that live in the `lsoa` folder, as per https://github.com/npct/pct-outputs-regional-R/commit/cb9f767773fda494de87d4590cef162ab32b221d . Also create corresponding rnet_full.geojson file for npct/pct-outputs-regional-notR- @Robinlovelace 
- [x] Make theMSOA layer 'route network lsoa clickable' option point to the  `lsoa/rnet.Rds` file... @usr110 OR copy lsoa/rnet.Rds to msoa/rnet.Rds
- [x] Update new regional rnet files @Robinlovelace 
- [x] Update the UI to allow users to select the Near Market scenario
- [x] Update the downloads page so that they point to the new rnets @AnnaGoodman1 
- [x] Update the other files in the national, R and not-R repos( @AnnaGoodman1  ), and move these to server (@nikolai-b )
- [x] Update region stats file when this issue resolved https://github.com/npct/pct-scripts/issues/83 - @AnnaGoodman1 
- [x] Generate national rasters - ~@mem48~ @Robinlovelace  tiles
- [x] Create tiles - @nikolai-b 
- [x] Test and deploy the updated data on test server @nikolai-b 
- [ ] update LA map? la-map.html, www/la-map-resources/la.js  @usr110  
- [x] Upload updated documentation @AnnaGoodman1 
- [ ] Deploy to production @nikolai-b 
- [ ] post blogs, link to them on front page @Robinlovelace @RachelAldred 

The person mentioned in each task is the best person for the job I think - let me know if you think any of these should be changed. A rough ETA of when these can be done would be useful (I can get the remaining data up by the end of the week). Looking forward to getting the updated national commute data out there!

Suggested target date for deployment: Friday 19th April, to provide 2 weeks for smoothing any wrinkles, but happy to push back if the tasks outlined above take longer.

**Comment by JDWoodcock** [comment]
thanks @Robinlovelace  what else needs to go out with near market in terms of documentation @RachelAldred and @AnnaGoodman1 . Our old result will change to- does anyone see any issues with this?

**Comment by Robinlovelace** [comment]
Good point about the documentation - I've added another point on that tagging Rachel for now.

**Comment by AnnaGoodman1** [comment]
@Robinlovelace , it looks like you have uploaded Isle of Wight data to the git server? please don't do that for other regions, otherwise you are doing a half launch, since that is the data the server is pointing to.

re. documentation, I think I have written what we need (At least a minimally acceptable version), currently here: Dropbox\PCT\2_WorkInProgress\Anna\NearMarket

**Comment by AnnaGoodman1** [comment]
ALSO please make all these shiny changes on the branch https://github.com/npct/pct-shiny/pull/668 !

**Comment by Robinlovelace** [comment]
Great. @AnnaGoodman1 these recent commits have no impact on the results because the new `rnet` files live in the MSOA not the LSOA folder. If you want to test the data, as I have done, you can do this:

```r
file.copy(paste0("../pct-outputs-regional-R/commute/lsoa/", r, "/rnet.Rds"),
          paste0("../pct-outputs-regional-R/commute/msoa/", r, "/rnet.Rds"), overwrite = TRUE)
```

But please revert that change and do not push it. Of course I'm not planning to do a 'half launch' but thanks for confirming! Note: the production server only updates when we change the 'data sha'.

Good point about making the changes on that branch. @usr110 are you up for testing that on the test LSOA commute rnet data that I've pushed for `isle-of-wight` for testing purposes? Good to keep the ball rolling on this. Working on the rasters.

**Comment by Robinlovelace** [comment]
BTW many thanks for updating the 'to-do list' @AnnaGoodman1, just noticed that. Very useful.

**Comment by AnnaGoodman1** [comment]
@Robinlovelace , oh yes , good point that they are lsoa files not previously there -phew!

**Comment by AnnaGoodman1** [comment]
@Robinlovelace , I am updating region stats info page. Where can I find info on what the minimum go dutch route segment number will be for each region? (i.e. in schools layer was usually 10 but 100 in London)

**Comment by Robinlovelace** [comment]
Good question that raises another one: where should we save this data? In the main national build file, or in a new one. This time the values are based on an upper limit of 20,000 features (~2/3rds of the schools rnet in London which is quite slow) and I rounded to the nearest 10, resulting in these numbers so far:


```
   region_name dutch_slc_min n_row
1                        london           150 19841
2            greater-manchester            30 19387
3         liverpool-city-region             0 17648
4               south-yorkshire             0 13959
5                    north-east            20 19535
6                 west-midlands            30 19230
7                  warwickshire            NA    NA
8                west-yorkshire            NA    NA
9                      cheshire            NA    NA
10                   lancashire            NA    NA
11                   humberside            NA    NA
12              north-yorkshire            NA    NA
13                   derbyshire            NA    NA
14               leicestershire            NA    NA
15              nottinghamshire            NA    NA
16       hereford-and-worcester            NA    NA
17                   shropshire            NA    NA
18                staffordshire            NA    NA
19                         avon             0 13829
20                        devon            NA    NA
21                       dorset            NA    NA
22                    wiltshire            NA    NA
23               cambridgeshire            NA    NA
24                 bedfordshire            NA    NA
25                        essex            NA    NA
26                         kent            NA    NA
27                    berkshire            NA    NA
28              buckinghamshire            NA    NA
29                  east-sussex            NA    NA
30                    hampshire            NA    NA
31                isle-of-wight             0  1311
32 cornwall-and-isles-of-scilly            NA    NA
33                      cumbria            NA    NA
34              gloucestershire            NA    NA
35                hertfordshire            NA    NA
36                 lincolnshire            NA    NA
37                      norfolk            NA    NA
38             northamptonshire            NA    NA
39               northumberland            NA    NA
40                  oxfordshire            NA    NA
41                     somerset            NA    NA
42                      suffolk            NA    NA
43                       surrey            NA    NA
44                  west-sussex            NA    NA
45                        wales            NA    NA
```

**Comment by AnnaGoodman1** [comment]
@Robinlovelace : As for schools layer, the current bodge is that I paste the numbers into the building stats files, i.e. https://github.com/npct/pct-scripts/blob/master/commute/lsoa/build_params_pct_region.csv . You ideally could set up the script to export this number into that file....? (Or we can leave it as a bodge.)

**Comment by Robinlovelace** [comment]
Thanks, yes that's the ideal solution, and I'll endeavour to add columns with the appropriate info. Many thanks!

**Comment by AnnaGoodman1** [comment]
example code is e.g. 
  build_params[build_params$region_name == region, ]$n_flow_rnet <- nrow(rf_rnet)

in 07b.2, line 45

**Comment by Robinlovelace** [comment]
I'll be able to add them all in one go soon: the regional rnets are half way done (for the second time: the first time I missed the ebike scenario).

**Comment by Robinlovelace** [comment]
Update on this: the national rnet has been uploaded! See: https://github.com/npct/pct-scripts/releases

**Comment by Robinlovelace** [comment]
@mem48 please can you do the rasterisation?

Update on methods: I suspect gdal may be faster than the previously used method - see here for documentation: https://gdal.org/gdal_rasterize.html

**Comment by AnnaGoodman1** [comment]
Ok great, either send me the full CSV or update the build Param file. Thanks Anna

**Comment by usr110** [comment]
> Good point about making the changes on that branch. @usr110 are you up for testing that on the test LSOA commute rnet data that I've pushed for `isle-of-wight` for testing purposes? Good to keep the ball rolling on this. Working on the rasters.

Sure. Will have to be next week - same goes for the the clickable `rnet` option. 
Glad you and @AnnaGoodman1 mentioned the branch. It was slightly confusing otherwise.

**Comment by Robinlovelace** [comment]
Hi guys, just found a faster way of generating the national rasters and the results are looking good! Question: what resolution? The results at the highest resolution (10 m) no longer look great because the new method doesn't convert lines into polygons before rasterizing. I think the polygons were ~30 wide in any case so I've tried rastering them at this resolution and I think the results look good. There are no more high value 'artefacts' at junctions. The results are more 'blocky' but I think this may be no bad thing, as we want people to understand that the results are not saying 'build on this specific path' but 'build along this desire line and in these areas'. Result below show before, after (30m) and after (10m). Cc @AnnaGoodman1 @JDWoodcock @RachelAldred @mem48 @nikolai-b @usr110 - thoughts welcome!

Another thought I had: we could make the visible raster values semi-transparent so they don't hide road names etc. I think that would be beneficial for users if not too hard to do.

Before:

![image](https://user-images.githubusercontent.com/1825120/55776575-23dfb080-5a95-11e9-9036-2c7354bea2d1.png)

After (30 m):

![image](https://user-images.githubusercontent.com/1825120/55776493-e0854200-5a94-11e9-84ac-2fdac9fdfade.png)

After (10 m):

![image](https://user-images.githubusercontent.com/1825120/55776642-55587c00-5a95-11e9-9826-841aa56b9444.png)

**Comment by Robinlovelace** [comment]
Also progress on getting the stats for the regional rnets, apologies this is slow in coming @AnnaGoodman1 : https://github.com/npct/pct-scripts/blob/755ff49728409b598e69b95570885631c3e109f0/commute/lsoa/build_params_pct_region.csv.

Any objections if I upload the lsoa route networks? Want to keep momentum up on this. Any update on the UI changes that can support this new data (already there for the test region of `isle-of-wight`)?

I think @usr110 or @nikolai-b are well placed to do the UI changes and that sooner is probably better than later from a testing perspective.

**Comment by RachelAldred** [comment]
Just a heads up that I am demonstrating the pct in Maidstone Tuesday afternoon at an event so would be good not to be doing anything then that might cause instability :) thanks.

**Comment by RachelAldred** [comment]
That's Tues 15th.

**Comment by Robinlovelace** [comment]
UI changes will only affect the test server so shouldn't be an issue. Thanks for heads-up in any case. Still aiming for deployment on 19th, but if UI/other changes are not ready by then we can always delay.

**Comment by AnnaGoodman1** [comment]
For raster resolution, can we not use the same method that was recently used in the schools layer? That looks very good, see below. I don't like the look of the 30m one, it is much less attractive than the current commute or schools layers

![Untitled](https://user-images.githubusercontent.com/11818225/55783857-45e22e80-5aa7-11e9-8137-fc8f1980c86b.png)

**Comment by AnnaGoodman1** [comment]
@usr110 , @Robinlovelace : In terms of the UI, , Robin and I had a conversation about this. My suggestion was that the lsoa rnet RDS files simply get saved in the relevant place in the msoa file structure too. I appreciate it is not DRY ideal to save the same file in two places, but that would mean that we would need pretty much no changes to pct-shiny code.  I think the alternative may be to introduce lots of messy 'if' statements into pct-shiny server.R file. I'm very happy to be proved wrong on this, but if it does look like it will be a coding nightmare then I suggest we do consider saving the LSOA file in two places

**Comment by Robinlovelace** [comment]
Hi @AnnaGoodman1 yes it's possible, but perhaps not within the time-scales required for the national build. Agree that image, which uses polygons representing roads, looks good. @mem48 tried to re-build it but lacks computer and perhaps time capacity to do it within the time constraints I believe. Can have a play. I think the optimum would be:

- To have high (10 m) underlying resolution
- To have partial transparency, so that any road names currently hidden in the image pasted above can be seen
- To remove the artefacts which provide incorrect results at junctions

**Comment by Robinlovelace** [comment]
Update, especially for @AnnaGoodman1 and @mem48, I've found a solution that hits all of those criteria, building on the code written by Malcolm, but using GDAL rather than `fasterize` to do the actual rasterisation, which seems much (100+ times) faster. 

Result (note the transparency, I think this is possible in leaflet) below. Thanks for the feedback, which has led to useful iteration.

![image](https://user-images.githubusercontent.com/1825120/55795061-cf512b00-5abe-11e9-94b7-9b61055afbb5.png)

**Comment by Robinlovelace** [comment]
Note: this is with the previous method, with artefacts. I think it's best to follow #704 and update the school raster rather than make the new commute raster like the old school raster based on this, and the fact it's quicker to produce:

![image](https://user-images.githubusercontent.com/1825120/55795338-76ce5d80-5abf-11e9-85ec-a4ff16232046.png)

**Comment by Robinlovelace** [comment]
A better image of the old raster file: 
![image](https://user-images.githubusercontent.com/1825120/55795418-a3827500-5abf-11e9-8173-030703098ef1.png)

**Comment by AnnaGoodman1** [comment]
newer method still looks as if it is not quite as nice as the previous one, but if the older method too intensive to be feasible to produce (you agree, @mem48 ?) then new method probably good enough,..

I am not totally convinced by partial transparency, the yellow and grey of most roads are slightly hard to see even when opaque on pct-bike. You aren't using the standard colour scheme I think @Robinlovelace , but if you apply it and add transparency I am not sure will be great...

**Comment by JDWoodcock** [comment]
thanks could we do a bit more testing so we can see perhaps for a region on test server?

**Comment by Robinlovelace** [comment]
And here's a before/after in central London over a similar area (ignore colours for now, although I think the new one will be less multicoloured for a given segment):

![image](https://user-images.githubusercontent.com/1825120/55806914-24019f80-5ad9-11e9-9c13-347a834ec72f.png)

**Comment by Robinlovelace** [comment]
![image](https://user-images.githubusercontent.com/1825120/55807004-56130180-5ad9-11e9-864b-13ea6191bbdd.png)

**Comment by AnnaGoodman1** [comment]
thanks @Robinlovelace , this new one is missing some segments, eg round the  strand - covers fewer of the roads, which seems less good to me. Are the rasters based on all trips in the vector rnet, not just those with >150 cycles? if so, why the gaps?

**Comment by Robinlovelace** [comment]
Not sure if this answers your question, but could it be the colourscheme used? If you open the file in QGIS and play with the Symbology, you can make all layers have a colour. I suspect it was because low values were white, which seemed invisible in the map, that they could not be seen. So below I've updated the colourscheme, that I think can help answer the question (and shows that in this updated version road segments will be less multicoloured and bity post tiling):

![image](https://user-images.githubusercontent.com/1825120/55835224-87a8be80-5b13-11e9-96eb-8393f3e576b2.png)

**Comment by Robinlovelace** [comment]
> Are the rasters based on all trips in the vector rnet, not just those with >150 cycles? 

No this is based on the full route network. Please try opening the raster result in QGIS - it clearly shows cycle use nationwide, not just in 'high cycling' areas in London.

**Comment by AnnaGoodman1** [comment]
![Untitled](https://user-images.githubusercontent.com/11818225/55856683-eea19400-5b62-11e9-8a1d-db475a273910.png)
I agree with full colours looks better, but as circled in red on the left there still seem to be some routes in old version not new? don't have time to do interactive mapview of rf_shape for this are (off on holiday rest of week), but could you @Robinlovelace  to check they are not wrongly missing? thanks!

**Comment by RachelAldred** [comment]
Yes there do seem to be gaps in London, e.g. Regent Street also seems to have no values (not even 0). Had a quick look at Preston and seemed less gappy there (except slight weirdness where people are routed down the Broughton bypass rather than the old main road, but I can imagine that it might count as a faster route for commutes.)

**Comment by Robinlovelace** [comment]
Can you compare the result with this dataset: https://github.com/npct/pct-scripts/releases/download/0.0.1/rnet_all.gpkg ? It may be that no routes went that way.

**Comment by RachelAldred** [comment]
Turns out that yes no one is routed down that bit of Regent Street, oddly enough. However for the Preston example, the gpkg file does cover Whittingham Lane at the Broughton crossroads, but it's missing for the raster.
![image](https://user-images.githubusercontent.com/11818218/55859346-c8cbbd80-5b69-11e9-8709-ea9411fe4382.png)

**Comment by Robinlovelace** [comment]
> for the Preston example, the gpkg file does cover Whittingham Lane at the Broughton crossroads, but it's missing for the raster.

Can you find the value of the `bicycle` field along this segment?

**Comment by Robinlovelace** [comment]
Aha - just saw from your image @RachelAldred that it is 0 - hence why it is not in the raster I think. Thanks for all the testing!

![image](https://user-images.githubusercontent.com/1825120/55860355-f31e7a80-5b6b-11e9-8fe1-fcee4769bae2.png)

**Comment by RachelAldred** [comment]
Ah ok sorry - I forgot zeroes would be missing from the raster. I'm at a workshop today till late afternoon, but I'll do some more checking between the two networks towards the end of the day (after having made zeroes invisible in the vector network).

**Comment by Robinlovelace** [comment]
Update: as discussed with @nikolai-b today, the plan is to trial the new LSOA raster layer whilst keeping the old tiles online for testing. Only after testing on the test server we will deploy, hopefully by/on **19th April**. Will tag other people when we need people to test. In any case, here are the new raster tiles, weighing in at nearly 40 GB unzipped: https://github.com/npct/pct-scripts/releases/download/0.0.1/ras_all_new_commute_10.zip

**Comment by RachelAldred** [comment]
I am on the train so cannot download and test 40GB of new tiles today. I will look tomorrow afternoon instead.

**Comment by Robinlovelace** [comment]
Great, thanks Rachel. It's around 250 MB to download. We'll have a chance to inspect these without access to tonnes of RAM when the raster tiles are built by @nikolai-b, over the weekend if things go well.

**Comment by mem48** [comment]
@Robinlovelace great work on getting this done, My PC is still committed for at least 3 more days. :(

**Comment by Robinlovelace** [comment]
Thanks for the positive message @mem48: your services, and your computers, will no doubt come in handy soon on this project once more ; )

**Comment by nikolai-b** [comment]
@Robinlovelace to keep communication simple I'm going to move to https://github.com/npct/pct-lsoa-vis/issues/1 to update on the tiling progress, then I'll post back here when done.

**Comment by Robinlovelace** [comment]
Hi all, especially @AnnaGoodman1 @JDWoodcock and @RachelAldred we're making good progress with this. @nikolai-b is building the tiles and plans to have them deployed by this evening (please let us know how you get on in any case Nikolai). After some testing by Nikolai and I on Tuesday, we should be ready for everyone to test this new data (and the memory fix by Nikolai) on Weds/Thurs, in time for a Friday deploy. It will be tight, but I think it's worth going for. I've added the LSOA route network data to the ouptuts-regional-R folder and plan to add the files to the not-R folder later today.

**Comment by Robinlovelace** [comment]
Major update on this: the new tiles are ready! @usr110 please can you test the latest version on the `new-ras` branch? See below for results in `isle-of-wight` locally - looking really good IMO:

![image](https://user-images.githubusercontent.com/1825120/56120878-fb5d2800-5f66-11e9-88b7-64ecdabe0d83.png)

See https://github.com/npct/pct-shiny/commit/8cdae97cdc4ec210f08f2ce00c2834136defbc2f

**Comment by Robinlovelace** [comment]
Go Dutch, London: 
![image](https://user-images.githubusercontent.com/1825120/56121267-aa016880-5f67-11e9-84d9-c8d0ff60b435.png)

**Comment by AnnaGoodman1** [comment]
@usr110 / @Robinlovelace , I think second point on list not resolved. Test it locally on isle of wight, making sure you have latest version of regional-files-R repo (i.e. no file existing called 'msoa/rnet.Rds). As per previous comment, I wonder if easiest just to have one rnet file saved in two places....

**Comment by usr110** [comment]
Sorry I didn't get a chance to get involved in this, so I am sort of out of loop here. I see that many new developments have been made, so great work everyone. However in terms of UI and testing specifically, I am not sure which branch should I use? Is it `new-ras` or `master` branch instead? @Robinlovelace 

> @usr110 / @Robinlovelace , I think second point on list not resolved. Test it locally on isle of wight, making sure you have latest version of regional-files-R repo (i.e. no file existing called 'msoa/rnet.Rds). As per previous comment, I wonder if easiest just to have one rnet file saved in two places....

As I mentioned above @AnnaGoodman1, I haven't done anything in this regard. I will get into it, if I know where (which branch) the work needs to be done. Saw @nikolai-b did some of the tasks originally assigned to me, but not sure if he has done anything about it.

**Comment by Robinlovelace** [comment]
Hi @usr110 please work on the master branch. Since @AnnaGoodman1 merged the near market branch, and I merged the `new-ras` and `memory` branches after testing last night, all the latest changes are on `master`. Thanks for coming back on board on this and if you could fix the second point that would be great: 

> Make theMSOA layer 'route network lsoa clickable' option point to the lsoa/rnet.Rds file... @usr110 OR copy lsoa/rnet.Rds to msoa/rnet.Rds

**Comment by usr110** [comment]
On it and thanks @Robinlovelace for the summary. Just noticed that I see `All trips` for `Isle-of-wight` - which is not right. Also shouldn't I now expect to see the `government target (new market)` in the `scenario` box? I see this now:

![image](https://user-images.githubusercontent.com/10958628/56202296-a4288780-603a-11e9-8422-5a6c95ce0e39.png)

Have locally updated the `pct-outputs-regional-R`.

**Comment by Robinlovelace** [comment]
Hi @usr110 no, I would only expect to see that in Manchester. I think it will not be there on the production branch but if you can remove it for other regions on the testing branch, that would be great. Cheers for testing - linked to #706. If you could fix that and point to the `rnet.Rds` in the `lsoa` folder that would be great - preferably via a Pull Request.

**Comment by usr110** [comment]
Have fixed the #706. Now `all trips` won't appear if the data directory doesn't exist (in addition to it never appearing on the `prod` server)

> Also shouldn't I now expect to see the government target (new market) in the scenario box?

Re my above comment about `scenario`, sorry I was mistaken. `Government Market (near market)` was already there.

![image](https://user-images.githubusercontent.com/10958628/56205598-2b2d2e00-6042-11e9-8b26-7af1a0215d77.png)

**Comment by Robinlovelace** [comment]
Cheers for the update @usr110. Any ideas how to fix the 'infinite spinner' issue (the loading spinner keeps going after selecting some options in the UI)?

**Comment by usr110** [comment]
> Any ideas how to fix the 'infinite spinner' issue (the loading spinner keeps going after selecting some options in the UI)?

Briefly looked at it. No idea how to resolve it. I think @nikolai-b is better placed for it.

> Make theMSOA layer 'route network lsoa clickable' option point to the lsoa/rnet.Rds file... @usr110 OR copy lsoa/rnet.Rds to msoa/rnet.Rds

For testing purposes I copied `pct-outputs-regional-R\commute\lsoa\isle-of-wight\rnet.Rds` to `pct-outputs-regional-R\commute\msoa\isle-of-wight\rnet.Rds`, and did nothing in the code, and the network start showing. So it's completed in a way. However I think we may need to show only selected geography specific `rnet`. For instance only show `LSOA (clickable)` or `LSOA (image)` when LSOA geography is selected - similarly for `MSOA`. What do you think? @Robinlovelace 

On a separate matter, I was looking at the `isle-of-wight` folder in the `pct-outputs-regional-R` repo, and found two files called `rnet` and `rnet_full`. How are they different? If `rnet_full` has all the routes (without any threshold), then shouldn't it be different? Just looked at their sizes, and both are of equal size. Also checked if they are identical, using

> identical(readRDS("commute/lsoa/isle-of-wight/rnet.Rds"), readRDS("commute/lsoa/isle-of-wight/rnet_full.Rds"))
[1] TRUE

**Comment by Robinlovelace** [comment]
Hi @usr110 regarding your question

> I think we may need to show only selected geography specific rnet. For instance only show LSOA (clickable) or LSOA (image) when LSOA geography is selected - similarly for MSOA. What do you think?

We're shifting to only providing the LSOA route network. So the labels in the dropdown menu can change, to be like (or the same as) the route network labels in the schools.

> On a separate matter, I was looking at the isle-of-wight folder in the pct-outputs-regional-R repo, and found two files called rnet and rnet_full. How are they different? If rnet_full has all the routes (without any threshold), then shouldn't it be different?

As with the schools rnets, we need 2 files: one for downloads and one for visualising. In small regions they are the same as you say. For more detail on this decision, see this comment, and others, in this thread: https://github.com/npct/pct-shiny/issues/676#issuecomment-470038955

**Comment by AnnaGoodman1** [comment]
the labels are currently correct but should be the LSOA rnet in both MSOA and LSOA layer. Unlike @Robinlovelace , I favour just copying the lsoa rnet.Rds to the msoa fodler, then as @usr110 says it works perfectly. Otherwise I suspect codebase changes may be quite fiddly (choose the selected 'geography' except if you are looking for the rnet in which case force it to be 'lsoa'...). But @Robinlovelace disagrees on this, and I am happy to be proved wrong & avoid having same file in 2 places.  (as @usr110 points out, we do also duplicate to rnet_full.Rds for small regions)

**Comment by Robinlovelace** [comment]
I'm confident that ignoring `geography` and hardcoding the LSOA folder for the vector `rnet` will be less work and better practice than copying all those MB of data. Also reduces the chances of confusion among people who find the data repos. So yes, fairly strong preference. I'm happy to do this. Heads-up also @nikolai-b.

**Comment by usr110** [comment]
Thanks both @Robinlovelace and @AnnaGoodman1 for the insight. I understand the issue of `rnet` for `MSOA` much better now. I was siding with you on @AnnaGoodman1 that changing the code at the `shiny` end would be rather fiddly, instead of easier way out of copying `rnet` from `LSOA` to `MSOA`. However it turns out it was rather simple. I made the changes here: https://github.com/npct/pct-shiny/commit/ca2dea5403cfcaf9340f7881ed971317d5e1aba5

With a bit of testing, it seems to work fine. You may like to test it yourself.

Re `rnet_full`, as you say that @Robinlovelace that for some of the smaller regions it will be identical to `rnet`. In that case, I would create a small file, say called `identical`,  indicating that it's identical to `rnet`, so `rnet` should be read instead.
For bigger regions, I would create separate `rnet` and `rnet_full`. This is more efficient, I think. What do you think?

**Comment by Robinlovelace** [comment]
Hi @usr110 great solution in https://github.com/npct/pct-shiny/commit/ca2dea5403cfcaf9340f7881ed971317d5e1aba5, I guessed it would be few lines of code and think the simple `if()` statement is a future-proof fix that adheres to the existing structure of the data. Fantastic work!

**Comment by Robinlovelace** [comment]
Regarding the separation of `rnet` and `rnet_full`, I agree with your idea in principle: even more DRY. However, given that the data has already been pushed, I'm not sure that implementing the idea is a good use of time. Happy to consider a PR that implements your suggestion if you can find a very fast fix. One other idea: use symbolic links to avoid unnecesary duplication of data. Question for @nikolai-b would symbolic links work like that in `git-lfs`?

**Comment by AnnaGoodman1** [comment]
Part of the issue is that rnet_full is accessed in the data download page. So you would need also to implement some statement there, saying 'do rnet full if exists, otherwise rnet'.  Fine if not fiddly, otherwise also fine by me to leave as is. Well done @usr110 (and @Robinlovelace )

**Comment by Robinlovelace** [comment]
Good point Anna, I was thinking only from `shiny`'s narrow perspective on the world :computer:. From the downloads perspective, it makes sense to keep them as separate files IMO.

**Comment by nikolai-b** [comment]
@Robinlovelace and @usr110 instead of `identical` just use [symbolic links](https://en.wikipedia.org/wiki/Symbolic_link). In linux `ln -s rnet.Rds rnet_full.Rds` then there is no duplication or extra code.

**Comment by Robinlovelace** [comment]
Good to know it's possible to use symbolic links with lfs. Not a priority though so I suggest we keep it as is for now.

**Comment by Robinlovelace** [comment]
Now that #710 is fixed thanks to uploaded data from @AnnaGoodman1 I think that is everything. Am I right in thinking that we just need to test the new data on the test server before deploying? I've checked in a few random regions and all seems good, and I think that #708 makes the data load quicker too, great work @nikolai-b. Please @JDWoodcock and @RachelAldred have a play and report back if you find any issues. This is almost ready to go!

**Comment by JDWoodcock** [comment]
@Robinlovelace what specifically should we check

**Comment by Robinlovelace** [comment]
Please check each of the scenarios works for zones, lines, routes, route network and route network LSOA options for a selection of regions, e.g.:

- [x] Isle of Wight
- [ ] London
- [ ] GM
- [ ] Cheshire
- [ ] Liverpool City Region
- [ ] West Midlands
- [ ] Devon

I think if we also try jumping between regions and type of trip that would be good - let us know of any bugs here. And of course any more issues with the landing page.

I've a few more ideas on next steps, discussed with @mem48 (putting the raster tiles on the landing page national map and, more tentatively, making the LSOA route network more prominent, maybe even on by default in the app) but will save that for another day.

**Comment by JDWoodcock** [comment]
some quick testing in central England and working well with minimal lags  :)
only issue is colouring of image route network. With tiles on it is often hard to make out. I think we need stronger colours. That said at the moment the clickable network is working pretty well- a few second delay when looking at whole of WM but not bad at all. I tested with chrome.

**Comment by AnnaGoodman1** [comment]
working well for me. Agree image rnet only looks v. good with black and white background and no zones - but that seems ok to me?

**Comment by Robinlovelace** [comment]
I think the fact that the basemap is B+W by default saves it. Disadvantages of disrupting user experience by changing the colourscheme outweigh any advantages of selecting stronger colours is my thinking. Great to hear there are no bugs, and great to hear it's nippy!

**Comment by Robinlovelace** [comment]
One question @AnnaGoodman1 have the downloadable geojsons been updated? Looking here: https://github.com/npct/pct-outputs-regional-notR

**Comment by RachelAldred** [comment]
Looked at London. The image rnet looks generally good to me - I like the way it gives a picture of where the various little feeder routes are coming from, by contrast the clickable route network looks quite disconnected in NE Outer London even at 90% (as you are not seeing all those joining streets that merge to make the higher numbers). Super exciting to have the downloadable LSOA vector network of course so people can visualise more if they want :)
It's sometimes hard to interpret the colours on the image - something I assumed was 1000-1999 in the image layer turned out to be 540 in clickable. But I think that's probably inevitable result of different monitors etc. London is a bit slow as expected with the amount of data but only one actual crash. I think it is ok.
We need to decide when the two blogs go up and I need to tweak the near market one and put on wordpress (thanks for comments).

**Comment by AnnaGoodman1** [comment]
@Robinlovelace , geojsons in progress - about 50% done after 30 hours. plus some national ones to do. hopefully all done by end of lond weekend

**Comment by mem48** [comment]
Perhaps we should colour code the clickable route network to match the image, might make comparison easier?

**Comment by AnnaGoodman1** [comment]
all new files now uploaded to git from me

**Comment by Robinlovelace** [comment]
> Perhaps we should colour code the clickable route network to match the image, might make comparison easier?

That's possible, happy to take a look how it looks, after the new data has been deployed though. Pull requests implementing this welcome for review/testing.

**Comment by Robinlovelace** [comment]
Now that the front page map has been fixed (see image below), great work @nikolai-b and the new files have all been uploaded I think this is ready to go. I suggest we test over the next few days and raise any further issues here, then merge `master` into `production` on Friday, test over the weekend and publicise the new data next week.

**Comment by Robinlovelace** [comment]
![image](https://user-images.githubusercontent.com/1825120/56636557-0dbf1c00-6661-11e9-955d-07552b4f2cbc.png)

No lines on Firefox, finally!

**Comment by AnnaGoodman1** [comment]
looking good on live server. @Robinlovelace , do you want to publish your blog and link to it from the whats new sextion of front page?

**Comment by Robinlovelace** [comment]
Yes will do, thanks @AnnaGoodman1 . The permalink is https://blogpctbike.wordpress.com/?p=360  - can add that but happy for you to do so also via PR. Thanks!
